src/TestingDevice.java

```diff
@@ -128,11 +145,20 @@ public Device(boolean isRandom) {
      * @param size the number of bits stored in this device
      * @param bitsPerPeek the number of bits to disclose via peek or set via poke
      */
-    public Device(int size, int bitsPerPeek) {
+    public TestingDevice(int size, int bitsPerPeek) {
         this.size = size;
         boolean[] initialBits = new boolean[this.size];
+        boolean allTrue = true;
+        boolean allFalse = true;
         for (int i = 0; i < size; i++) {
-            if (Math.random() > 0.5) initialBits[i] = true;
+            if (Math.random() > 0.5) { initialBits[i] = true; allFalse = false; }
```

Comment:
This would be better to read if this if statement was expanded to the appropriate lines. However, since we are not turning this file in I don't think it matters to much.

Reply:
I am fairly certain I try to avoid one line wrapped in brackets. I think IntelliJ does this sometimes due to its codestyle. I will look into it.

Changed in the bugfixes branch.